### PR TITLE
releng: Add jobs for the `release-1.30` branch

### PR DESCRIPTION
**Pull request by xmudrii**
Supersedes https://github.com/kubernetes/test-infra/pull/32321

This PR is based on #32321 with additional fixes to the version markers.

/assign @dims 
cc @kubernetes/release-engineering @mehabhalodiya 

/sig release
/area release-eng
/priority critical-urgent

### CODE REVIEW

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.30.yaml [hunk 1]
```diff
@@ -0,0 +1,2473 @@
...
+      preset-azure-anonymous-pull: "true"
+      preset-azure-capz-sa-cred: "true"
+      preset-azure-cred-only: "true"
+      preset-capz-containerd-1-7-latest: "true"
+      preset-capz-windows-2022: "true"
+      preset-capz-windows-common-pull: "true"
+      preset-dind-enabled: "true"
+      preset-kind-volume-mounts: "true"
+      preset-windows-private-registry-cred: "true"
+    name: pull-kubernetes-e2e-capz-windows-master
```
Comment (dims) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.30.yaml:2314]: one more .. `name: pull-kubernetes-e2e-capz-windows-master` should be name: `pull-kubernetes-e2e-capz-windows-1.30`

File: config/jobs/kubernetes/generated/generated.yaml [hunk 2]
```diff
@@ -20,15 +20,12 @@ periodics:
...
+      - --extract=ci/latest-1.30
       - --extract-ci-bucket=k8s-release-dev
       - --timeout=180m
       - --env=KUBE_PROXY_DAEMONSET=true
       - --env=ENABLE_POD_PRIORITY=true
-      # TODO: Fix generate_tests.py to set DisableCloudProviders,DisableKubeletCloudCredentialProviders to true when --cloud-proider=external
-      # And also add --env=ENABLE_AUTH_PROVIDER_GCP=true
-      - --env=KUBE_FEATURE_GATES=AllAlpha=true,InTreePluginGCEUnregister=false,DisableCloudProviders=true,DisableKubeletCloudCredentialProviders=true
-      - --env=ENABLE_AUTH_PROVIDER_GCP=true
+      - --env=KUBE_FEATURE_GATES=AllAlpha=true,InTreePluginGCEUnregister=false,DisableCloudProviders=false,DisableKubeletCloudCredentialProviders=false
```
Comment (Vyom-Yadav) [inline config/jobs/kubernetes/generated/generated.yaml:28]: @dims I have lost the context on this. But I found: https://github.com/kubernetes/kubernetes/pull/117503, can you please confirm if this is fine?

Comment (dims) [reply config/jobs/kubernetes/generated/generated.yaml:28]: yes this is fine.

**Pull request merged**